Automattic/studio
# Add Sync protocol listener to connect sites from web

## Related issues

- Fixes https://github.com/Automattic/dotcom-forge/issues/9359

## Proposed Changes

- Add deep link listener to open Studio and connect a site automatically
- It connects the remote site to the Studio site that started the flow

## Testing Instructions



- Follow the testing instructions of https://github.com/Automattic/wp-calypso/pull/96691
- Or access this URL from your computer. Replace the site ids: `wpcom-local-dev://sync-connect-site?studioSiteId=${ studioSiteId }&remoteSiteId=${ siteId }`
- Observe Studio opens and the remoteSiteId is connected to your current site.

Full flow:

https://github.com/user-attachments/assets/91caa573-4048-4416-b278-594de8766460

Just trigger the connect deep link:

Example URL: `/home/239281838?studioSiteId=026a8080-362a-4ab9-921d-2c63510ffe5a`

https://github.com/user-attachments/assets/9465efda-d727-4fd9-80a3-7bc314ee6c5c


## Pre-merge Checklist
